src/utils.js

```diff
@@ -82,9 +50,80 @@ const formatForbiddenLicenseError = licenses => {
   return `${header}\n${lines}`;
 };
 
+/**
+ * Generates a SPDX expression joining the input licenses with the OR operator
+ * @param {string[]} licenses - List of SPDX licenses
+ * @return {string} - A string with the input licenses joined with the OR operator.
+ * e.g. input -> ['MIT', 'GPL-1.0'] output -> 'MIT OR GPL-1.0'
+ */
+const generateSPDXExpression = (licenses = []) => licenses.join(' OR ');
+
+/**
+ * Checks whether the input licenses are SPDX compliant. Throws an error with the invalid licenses
+ * @param licenses - List of licenses
+ */
+const checkSPDXCompliance = (licenses = []) => {
+  const invalidLicenses = licenses.filter(arg => !isSPDXCompliant(arg));
+  if (invalidLicenses.length) {
+    throw new Error(
+      `The following licenses are not SPDX compliant. Please, use the --checkLicense option to validate your input:\n${invalidLicenses.join(' | ')}`
+    );
+  }
+};
+
+// @TODO Remove after issue has been solved
+/**
+ * Throws an error inf any the input licenses is a license affected by the spdx-satisfies issue.
+ * @param licenses - List of SPDX licenses
+ */
+const checkLicenseError = (licenses = []) => {
+  const errorLicenses = licenses.some(isLicenseError);
+  if (errorLicenses) {
+    throw new Error(
+      'Your failOn list contains a GFDL-1.x licenses and they are temporary unallowed. There\'s an issue pending to solve.'
+    );
+  }
+};
+
+// @TODO Remove after issue has been solved
+/**
+ * This is a temporal fix until an issue in spdx-satisfies package has been solved. The package itself throws an error if a GFDL license is used in the check: https://github.com/jslicense/spdx-satisfies.js/issues/15
+ * @param {string} license - A SPDX license
+ * @return {boolean}
+ */
+const isLicenseError = (license = '') => licensesExceptions.includes(license);
+
+/**
+ * Checks the SPDX expression against the list of packages. If the license of the package itself is not SPDX compliant, the package will
+ * be included on the "nonCompliant" list. If the SPDX expression satisfies the package license, the package will be included on the
+ * "satisfied" list
+ * @param {string} expression - A SPDX expression
+ * @param {object[]} packages - A list of packages to be checked against the SPDX expression
+ * @return {{satisfied: object[], nonCompliant: object[]}} - A couple of lists including the packages that satisfy the SPDX expression
```

Comment:
We should double-check the JSDoc for this function after the latest changes. The return is now different, and probably the semantic of the function is too.

Reply:
Done